android/beagle/src/main/java/br/com/zup/beagle/android/cache/CacheManager.kt

```diff
@@ -68,21 +68,25 @@ internal class CacheManager(
 
         val beagleCache = getBeagleCacheFromDisk(url, beagleHashKey)
 
-        return if (beagleCache == null) {
-            null
-        } else if (beagleCache.isExpired()) {
-            deleteDiskCacheForUrl(url)
-            null
-        } else {
-            beagleCache
+        return when {
+            beagleCache == null -> {
+                null
+            }
+            beagleCache.isExpired() -> {
+                deleteDiskCacheForUrl(url)
+                null
+            }
+            else -> {
+                beagleCache
+            }
         }
     }
 
     private fun getBeagleCacheFromDisk(url: String, beagleHashKey: String): BeagleCache? {
         val beagleJsonKey = url.toBeagleJsonKey()
         val beagleTimeKey = url.toBeagleTimeKey()
 
-        val cachedData = storeHandler.restore(StoreType.DATABASE, beagleHashKey, beagleJsonKey, beagleTimeKey)
+        val cachedData = storeHandler!!.restore(StoreType.DATABASE, beagleHashKey, beagleJsonKey, beagleTimeKey)
```

Comment:
Is it possible to make this implementation without the `!!` ??

Reply:
No.

Reply:
There is always a possibility to replace !!. In this case, I suggest adding any exception other than `KotlinNullPointerException` with a message of what the user should do.
```suggestion
        val cachedData = storeHandler?.restore(StoreType.DATABASE, beagleHashKey, beagleJsonKey, beagleTimeKey) ?: throw NoClassDefFound("Please, provide a StoreHandler implementation")
```